File: MatriX/samples/csharp/Weather/Weather.cs
using Matrix.Xml;

namespace Weather
{
    public class Weather : XmppXElement
    {
        public Weather() : base("ag-software:weather", "weather")
        {
        }

        public int Humidity
        {
            get { return GetTagInt("humidity"); }
            set { SetTag("humidity", value); }
        }

        public int Temperature
        {
            get { return GetTagInt("temperature");}
            set { SetTag("temperature", value); }
        }

        public string Zip
        {
            get { return GetTag("zip"); }
            set { SetTag("zip", value); }
        }
    }
}

File: MatriX/samples/vb/Weather/Weather.vb
Imports Matrix.Xml

Public Class Weather
	Inherits XmppXElement
	Public Sub New()
		MyBase.New("ag-software:weather", "weather")
	End Sub

	Public Property Humidity() As Integer
		Get
			Return GetTagInt("humidity")
		End Get
		Set
			SetTag("humidity", value)
		End Set
	End Property

	Public Property Temperature() As Integer
		Get
			Return GetTagInt("temperature")
		End Get
		Set
			SetTag("temperature", value)
		End Set
	End Property

	Public Property Zip() As String
		Get
			Return GetTag("zip")
		End Get
		Set
			SetTag("zip", value)
		End Set
	End Property
End Class